### PR TITLE
CAL-38 update NitfStoragePlugin so it only runs on nitf content and multiple content items

### PR DESCRIPTION
**Pull request by glenhein**
#### What does this PR do?
Fix a bug in NitfPreStoragePlugin so the plugin only runs on nitf content and so the plugin handles multiple content items.
#### Who is reviewing it?
@dcruver @kcwire @jrnorth @rzwiefel 
#### How should this be tested?
1) Upload nitf content (sample content is in catalog/imaging/catalog-imaging-transformer/src/test/resources), verify that the overview is created
2) Upload non-nitif content, if debug logging is enable, there will be an entry: "skipping content item: filename={} mimeType={}"

#### What are the relevant tickets?
CAL-38
#### Checklist:
- [ ] Documentation Updated
- [X] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: catalog/imaging/catalog-imaging-transformer/src/main/java/com/connexta/alliance/transformer/nitf/NitfPreStoragePlugin.java [hunk 1]
```diff
@@ -95,12 +95,21 @@ private void process(List<ContentItem> contentItems) {
         ContentItem contentItem = contentItems.get(0);
         Metacard metacard = contentItem.getMetacard();
 
+        if (contentItem.getMimeTypeRawData() == null || !contentItem.getMimeTypeRawData()
```
Comment (jrnorth) [inline catalog/imaging/catalog-imaging-transformer/src/main/java/com/connexta/alliance/transformer/nitf/NitfPreStoragePlugin.java]: Could simplify this a bit:
`if (!NitfInputTransformer.MIME_TYPE.equals(contentItem.getMimeTypeRawData()))`

File: catalog/imaging/catalog-imaging-transformer/src/test/java/com/connexta/alliance/transformer/nitf/TestPreStoragePlugin.java [hunk 2]
```diff
@@ -79,6 +79,7 @@ public void setUp()
         when(contentItem.getMetacard()).thenReturn(metacard);
         when(contentItem.getId()).thenReturn("101ABC");
         when(contentItem.getInputStream()).thenReturn(getInputStream(GEO_NITF));
+        when(contentItem.getMimeTypeRawData()).thenReturn(NitfInputTransformer.MIME_TYPE);
```
Comment (jrnorth) [inline catalog/imaging/catalog-imaging-transformer/src/test/java/com/connexta/alliance/transformer/nitf/TestPreStoragePlugin.java]: We'll probably want to add a unit test for a content item that isn't a NITF.

File: catalog/imaging/catalog-imaging-transformer/src/main/java/com/connexta/alliance/transformer/nitf/NitfPreStoragePlugin.java [hunk 3]
```diff
@@ -95,12 +95,21 @@ private void process(List<ContentItem> contentItems) {
...
+                .equals(NitfInputTransformer.MIME_TYPE)) {
```
Comment (jlcsmith) [inline catalog/imaging/catalog-imaging-transformer/src/main/java/com/connexta/alliance/transformer/nitf/NitfPreStoragePlugin.java]: equalsIgnoreCase?

File: catalog/imaging/catalog-imaging-transformer/src/main/java/com/connexta/alliance/transformer/nitf/NitfPreStoragePlugin.java [hunk 4]
```diff
@@ -95,12 +95,21 @@ private void process(List<ContentItem> contentItems) {
...
+            LOGGER.debug("skipping content item: filename={} mimeType={}",
+                    contentItem.getFilename(),
+                    contentItem.getMimeTypeRawData());
+            return;
+        }
+
```
Comment (dcruver) [inline catalog/imaging/catalog-imaging-transformer/src/main/java/com/connexta/alliance/transformer/nitf/NitfPreStoragePlugin.java]: This seems like code that will need to be repeated in many PreStoragePlugins.  Can we move it to a parent class?

File: catalog/imaging/catalog-imaging-transformer/src/main/java/com/connexta/alliance/transformer/nitf/NitfInputTransformer.java [hunk 5]
```diff
@@ -59,7 +59,7 @@
 
     private static final String ID = "ddf/catalog/transformer/nitf";
 
-    private static final String MIME_TYPE = "image/nitf";
+    static final String MIME_TYPE = "image/nitf";
```
Comment (dcruver) [inline catalog/imaging/catalog-imaging-transformer/src/main/java/com/connexta/alliance/transformer/nitf/NitfInputTransformer.java]: Can we make this an instance of javax.activation.MimeType instead of String?

Comment (glenhein) [inline catalog/imaging/catalog-imaging-transformer/src/main/java/com/connexta/alliance/transformer/nitf/NitfInputTransformer.java]: I'd rather not do that now, since the change would go into ddf-2.10.0-SNAPSHOT and Alliance is set to ddf-2.9.0 release.

File: catalog/imaging/catalog-imaging-transformer/src/main/java/com/connexta/alliance/transformer/nitf/NitfPreStoragePlugin.java (hunk 4 above)
Comment (glenhein) [inline catalog/imaging/catalog-imaging-transformer/src/main/java/com/connexta/alliance/transformer/nitf/NitfPreStoragePlugin.java]: I'd rather not do that now, since the change would go into ddf-2.10.0-SNAPSHOT and Alliance is set to ddf-2.9.0 release.

File: catalog/imaging/catalog-imaging-transformer/src/main/java/com/connexta/alliance/transformer/nitf/NitfInputTransformer.java (hunk 5 above)
Comment (glenhein) [inline catalog/imaging/catalog-imaging-transformer/src/main/java/com/connexta/alliance/transformer/nitf/NitfInputTransformer.java]: The MimeType ctor throws an exception, so it would have to go into a static initializer block, but would the block do if it couldn't create the MimeType instance? Throw a runtime exception?

Comment (jrnorth) [inline catalog/imaging/catalog-imaging-transformer/src/main/java/com/connexta/alliance/transformer/nitf/NitfInputTransformer.java]: An [`ExceptionInInitializerError`](http://docs.oracle.com/javase/8/docs/api/java/lang/ExceptionInInitializerError.html) would be appropriate.

File: catalog/imaging/catalog-imaging-transformer/src/main/java/com/connexta/alliance/transformer/nitf/NitfPreStoragePlugin.java (hunk 4 above)
Comment (dcruver) [inline catalog/imaging/catalog-imaging-transformer/src/main/java/com/connexta/alliance/transformer/nitf/NitfPreStoragePlugin.java]: Understood.  We'll create a story to do this after the Alliance 2.9.0 (or 2.9.1) release, then?

**Comment by kcwire** [comment]
👍

File: catalog/imaging/catalog-imaging-transformer/src/main/java/com/connexta/alliance/transformer/nitf/NitfPreStoragePlugin.java [hunk 6]
```diff
@@ -91,44 +94,72 @@ public UpdateStorageRequest process(UpdateStorageRequest updateStorageRequest)
...
             if (renderedImage != null) {
                 addThumbnailToMetacard(metacard, renderedImage);
-                ContentItem overviewContentItem = createOverview(contentItem.getId(), renderedImage,
+                ContentItem overviewContentItem = createOverview(contentItem.getId(),
+                        renderedImage,
                         metacard);
 
                 if (overviewContentItem != null) {
-                    contentItems.add(overviewContentItem);
+                    return Optional.of(overviewContentItem);
```
Comment (dcruver) [inline catalog/imaging/catalog-imaging-transformer/src/main/java/com/connexta/alliance/transformer/nitf/NitfPreStoragePlugin.java]: Could you remove the conditional and just return 'Optional.ofNullable(overviewContentItem)'?

File: catalog/imaging/catalog-imaging-transformer/src/main/java/com/connexta/alliance/transformer/nitf/NitfPreStoragePlugin.java [hunk 7]
```diff
@@ -91,44 +94,72 @@ public UpdateStorageRequest process(UpdateStorageRequest updateStorageRequest)
...
+                .get(0);
 
         NitfRenderer renderer = new NitfRenderer();
-        ImageInputStream inputStream = new MemoryCacheImageInputStream(
-                new ByteArrayInputStream(image));
-        BufferedImage bufferedImage = renderer.render(header, inputStream);
-        return bufferedImage;
+        ImageInputStream inputStream = new MemoryCacheImageInputStream(new ByteArrayInputStream(
+                image));
+        return renderer.render(header, inputStream);
```
Comment (dcruver) [inline catalog/imaging/catalog-imaging-transformer/src/main/java/com/connexta/alliance/transformer/nitf/NitfPreStoragePlugin.java]: Why this change?

File: catalog/imaging/catalog-imaging-transformer/src/main/java/com/connexta/alliance/transformer/nitf/NitfPreStoragePlugin.java [hunk 8]
```diff
@@ -164,14 +199,19 @@ private ContentItem createOverview(String id, BufferedImage image, Metacard meta
 
     private String buildOverviewTitle(String title) {
         String rootFileName = FilenameUtils.getBaseName(title);
-        String overviewFileName = String.format(OVERVIEW_FILENAME_PATTERN, OVERVIEW, rootFileName, JPG);
-        return overviewFileName;
+        return String.format(OVERVIEW_FILENAME_PATTERN,
```
Comment (dcruver) [inline catalog/imaging/catalog-imaging-transformer/src/main/java/com/connexta/alliance/transformer/nitf/NitfPreStoragePlugin.java]: Why this change?

File: catalog/imaging/catalog-imaging-transformer/src/main/java/com/connexta/alliance/transformer/nitf/NitfPreStoragePlugin.java (hunk 7 above)
Comment (glenhein) [inline catalog/imaging/catalog-imaging-transformer/src/main/java/com/connexta/alliance/transformer/nitf/NitfPreStoragePlugin.java]: It was just a style warning from intellij.

File: catalog/imaging/catalog-imaging-transformer/src/main/java/com/connexta/alliance/transformer/nitf/NitfPreStoragePlugin.java (hunk 8 above)
Comment (glenhein) [inline catalog/imaging/catalog-imaging-transformer/src/main/java/com/connexta/alliance/transformer/nitf/NitfPreStoragePlugin.java]: It was just a style warning from intellij.

File: catalog/imaging/catalog-imaging-transformer/src/main/java/com/connexta/alliance/transformer/nitf/NitfPreStoragePlugin.java (hunk 7 above)
Comment (dcruver) [inline catalog/imaging/catalog-imaging-transformer/src/main/java/com/connexta/alliance/transformer/nitf/NitfPreStoragePlugin.java]: I prefer the assignment because when debugging you can see the final value before it's returned while still being in the stackframe.

File: catalog/imaging/catalog-imaging-transformer/src/test/java/com/connexta/alliance/transformer/nitf/TestPreStoragePluginMulti.java [hunk 9]
```diff
@@ -0,0 +1,158 @@
+package com.connexta.alliance.transformer.nitf;
```
Comment (jaymcnallie) [inline catalog/imaging/catalog-imaging-transformer/src/test/java/com/connexta/alliance/transformer/nitf/TestPreStoragePluginMulti.java]: Missing header.

**Comment by jaymcnallie** [comment]
Built and tested.  👍  pending header fix.
